components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/validators/LogoutReqSignatureValidator.java

```diff
@@ -0,0 +1,256 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.samlsso.logout.validators;
+
+import net.shibboleth.utilities.java.support.codec.Base64Support;
+import org.apache.commons.lang.StringUtils;
+import org.opensaml.core.criterion.EntityIdCriterion;
+import org.opensaml.security.SecurityException;
+import net.shibboleth.utilities.java.support.net.URISupport;
+import net.shibboleth.utilities.java.support.resolver.CriteriaSet;
+import org.opensaml.security.x509.X509Credential;
+import org.opensaml.xmlsec.config.DefaultSecurityConfigurationBootstrap;
+import org.opensaml.security.credential.impl.CollectionCredentialResolver;
+import org.opensaml.security.credential.Credential;
+import org.opensaml.security.credential.UsageType;
+import org.opensaml.security.criteria.UsageCriterion;
+import org.opensaml.xmlsec.keyinfo.KeyInfoCredentialResolver;
+import org.opensaml.xmlsec.signature.support.SignatureException;
+import org.opensaml.xmlsec.signature.support.SignatureTrustEngine;
+import org.opensaml.xmlsec.signature.SignableXMLObject;
+import org.opensaml.xmlsec.signature.support.SignatureValidator;
+import org.opensaml.xmlsec.signature.support.impl.ExplicitKeySignatureTrustEngine;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authenticator.samlsso.manager.X509CredentialImpl;
+import org.wso2.carbon.identity.base.IdentityException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.SIGNATURE;
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.SIGNATURE_ALGORITHM;
+import static org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants.HTTP_POST_PARAM_SAML2_AUTH_REQ;
+import static org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants.RELAY_STATE;
+
+/**
+ * This class is used to validate the signature in SAML logout Request.
+ */
+public class LogoutReqSignatureValidator {
+
+    private static final Log log = LogFactory.getLog(LogoutReqSignatureValidator.class);
+
+    /**
+     * Validates the signature of the SAML requests sent with HTTP Redirect Binding against the given certificate.
+     *
+     * @param queryString SAML request (passed an an HTTP query parameter).
+     * @param issuer      Issuer of the SAML request.
+     * @param certificate Certificate for validating the signature
```

Comment:
Certificate for validating the signature.

Reply:
Fixed with PR https://github.com/wso2-extensions/identity-outbound-auth-samlsso/pull/133